test/new-e2e/tests/agent-platform/common/utils.go

```diff
@@ -0,0 +1,170 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	e2eClient "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	pkgmanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/pkg-manager"
+	svcmanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/svc-manager"
+	"gopkg.in/yaml.v2"
+)
+
+// ServiceManager generic interface
+type ServiceManager interface {
+	Status(service string) (string, error)
+	Start(service string) (string, error)
+	Stop(service string) (string, error)
+	Restart(service string) (string, error)
+}
+
+// PackageManager generic interface
+type PackageManager interface {
+	Remove(pkg string) (string, error)
+}
+
+// FileManager generic interface
+type FileManager interface {
+	ReadFile(path string) (string, error)
+	CheckFile(path string) (string, error)
+	FindFileInFolder(path string) (string, error)
+	WriteFile(path string, content string) (string, error)
+}
+
+// Helper generic interface
+type Helper interface {
+	GetInstallFolder() string
+	GetConfigFolder() string
+	GetBinaryPath() string
+}
+
+func getServiceManager(vmClient *e2eClient.VMClient) ServiceManager {
+	if _, err := vmClient.ExecuteWithError("systemctl --version"); err == nil {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Alternative seen in installer tests is `command -v <command>`, seems to be more efficient as it does not run command code, it just checks for it to exist